crane/template.go

```diff
@@ -0,0 +1,131 @@
+package crane
+
+import (
+	"fmt"
+	"github.com/michaelsauter/crane/print"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+	"text/template"
+)
+
+type TemplateInfo struct {
+	TargetedContainers []ContainerInfo
+	DependencyMap      DependencyGraph
+	Groups             map[string][]string
+}
+type TemplateInfoForContainer struct {
+	CurrentContainer   ContainerInfo
+	TargetedContainers []ContainerInfo
+	DependencyMap      DependencyGraph
+	Groups             map[string][]string
+}
+
+// Sub-interface of Container
+type ContainerInfo interface {
+	Name() string
+	Dockerfile() string
+	Image() string
+	Id() string
+	Dependencies() *Dependencies
+	Exists() bool
+	Running() bool
+	Paused() bool
+	ImageExists() bool
+	Status() []string
+}
+
+// set of additional functions for the templates
+var funcMap = template.FuncMap{
+	"Contains":    strings.Contains,
+	"ContainsAny": strings.ContainsAny,
+	"HasPrefix":   strings.HasPrefix,
+	"Join":        strings.Join,
+	"Split":       strings.Split,
+	"ToLower":     strings.ToLower,
+	"ToTitle":     strings.ToTitle,
+	"ToUpper":     strings.ToUpper,
+	"MatchString": regexp.MatchString,
+}
+
+// returns a TemplateInfo from a Config object
+func ConfigToTemplateInfo(c Config) TemplateInfo {
+	containers := []ContainerInfo{}
+	for _, container := range c.TargetedContainers() {
+		containers = append(containers, container)
+	}
+	return TemplateInfo{containers, c.DependencyGraph(), c.Groups()}
+}
+
+// returns a TemplateInfoForContainer from a ContainerInfo and a container
+func (td TemplateInfo) forContainer(currentContainer ContainerInfo) TemplateInfoForContainer {
+	return TemplateInfoForContainer{currentContainer, td.TargetedContainers, td.DependencyMap, td.Groups}
+}
+
+// generates a textual output for the selected template and templateInfo
+func (templateInfo TemplateInfo) template(templatePath string, outputPath string) {
+	if templatePath == "" {
+		print.Errorf("You must select the input template with the parameter --template (-t)\n")
+		return
+	}
+	tmpl, err := template.New(path.Base(templatePath)).Funcs(funcMap).ParseFiles(templatePath)
+	if err != nil {
+		print.Errorf("Unable to parse template: %s\n", err)
+		return
+	}
+	if strings.Contains(outputPath, "%s") {
+		for _, container := range templateInfo.TargetedContainers {
+			containerOutputPath := fmt.Sprintf(outputPath, container.Name())
+			err = generateFile(tmpl, containerOutputPath, templateInfo.forContainer(container))
+			if err != nil {
+				return
+			}
+		}
+	} else {
+		generateFile(tmpl, outputPath, templateInfo)
+	}
+}
+
+// generates a file with the output. If outputPath is empty, it will be dumped to os.Stdout
+func generateFile(tmpl *template.Template, outputPath string, templateInfo interface{}) error {
```

Comment:
with a `panic(StatusError{fmt.Errorf(errorDescription, errorStatus})`? Which `errorStatus` can I use? There is any convention?